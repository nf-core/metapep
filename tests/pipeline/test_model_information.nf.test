nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_model_information"
    tag "pipeline"

    test("test model information subbranch") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },
                { assert snapshot(path("$outputDir/model_information/mhcflurry.v1.4.3.supported_alleles.txt"),
                                path("$outputDir/model_information/mhcflurry.v1.4.3.supported_lengths.txt"),
                                path("$outputDir/model_information/mhcnuggets-class-1.v2.3.2.supported_alleles.txt"),
                                path("$outputDir/model_information/mhcnuggets-class-1.v2.3.2.supported_lengths.txt"),
                                path("$outputDir/model_information/mhcnuggets-class-2.v2.3.2.supported_alleles.txt"),
                                path("$outputDir/model_information/mhcnuggets-class-2.v2.3.2.supported_lengths.txt"),
                                path("$outputDir/model_information/syfpeithi.v1.0.supported_alleles.txt"),
                                path("$outputDir/model_information/syfpeithi.v1.0.supported_lengths.txt")).match("model_information") }
            )

        }

    }

}
