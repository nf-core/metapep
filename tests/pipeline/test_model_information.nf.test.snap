{
    "model_information": {
        "content": [
            "mhcflurry.v1.4.3.supported_alleles.txt:md5,1085d0010de0212a5d50d1639ccb8a0a",
            "mhcflurry.v1.4.3.supported_lengths.txt:md5,5cff92e8752af1f00ee27f91f6ce6560",
            "mhcnuggets-class-1.v2.3.2.supported_alleles.txt":"a322904c2327898c4a2e4b1824ef2efb",
            "mhcnuggets-class-1.v2.3.2.supported_lengths.txt":"24ef44781f5012a27121050365845185",
            "mhcnuggets-class-2.v2.3.2.supported_alleles.txt":"1f0230520650459480b5e41e280609d5",
            "mhcnuggets-class-2.v2.3.2.supported_lengths.txt":"05340a7f5c43c23908f16ab5929b3481",
            "syfpeithi.v1.0.supported_alleles.txt":"e007149d8f41dafc1290791c5f726194",
            "syfpeithi.v1.0.supported_lengths.txt":"09d2bfbc74ac9aa5bb1a650f02b848df"
        ],
        "timestamp": "2024-06-26T08:14:54+0000"
    }
}
